### ISSUE TITLE
undefined is not a function(evaluation 'object.getOwnPropertyDescriptors(arguments[i])')

### ISSUE THREAD

**Issue by azcodia** [facebook/react-native#23092]
before it went smoothly, but after I made native react a new error occurred,
this is the problem
"undefined is not a function (evaluation 'object.getOwnPropertyDescriptors (arguments [i])')"

even though the code is not I add a little bit of input
and that makes me crazy

**Comment by punyasurya** [comment]
have the same issue here...

**Comment by thaitrandng** [comment]
have the same issue here... =))

**Comment by luolin2611** [comment]
Same problem

**Comment by gabsima-nexapp** [comment]
I've updated from react native 0.55 to 0.57 and i get this error on android while on ios everything works fine.

I've updated the dependencies in gradle in order to be able to build but i still get the error..

**Comment by truongluong1314520** [comment]
please re-open this issue, I have the same problem like them :(

**Comment by rodolphefauquez** [comment]
Same problem here, android only, without reason (it was working this morning for me).
But the error disappear if i enable the 'debug JS remotely' option...

**Comment by luolin2611** [comment]
Same problem

![image](https://user-images.githubusercontent.com/17327008/51511350-ee370e00-1e3b-11e9-8ff8-e4692019b8ca.png)

**Comment by truongluong1314520** [comment]
@luolin2611 , that is it. I create a complete new project but the error appear at the first time running the project.

**Comment by mikecobas** [comment]
same problem

**Comment by loves127** [comment]
same problem

**Comment by luolin2611** [comment]
> @luolin2611 , that is it. I create a complete new project but the error appear at the first time running the project.



> .

I also run the error for the first time, and I can’t go.

**Comment by congnt148** [comment]
Same problem

**Comment by Army-92** [comment]
same problem

**Comment by melodion** [comment]
same problem

**Comment by JintoAntony** [comment]
Same issue here.

**Comment by liuqing520it** [comment]
restart your computer to solve this problem

**Comment by Iungad** [comment]
Same issue here.

**Comment by carlmagumpara** [comment]
same problem here

**Comment by PIlepRusel** [comment]
> restart your computer to solve this problem
does this fix the problem ?

**Comment by ken-talkingsource** [comment]
> > restart your computer to solve this problem
> > does this fix the problem ?

I restarted (my mac) but didn't solve this problem.

**Comment by huiruo** [comment]
react native undefined is not an object(evaluating 'Object.getOwnPropertyDescriptors(arguments[i])')

**Comment by aldy010** [comment]
still no fix yet, tried the above solution

**Comment by mohsincox** [comment]
I restarted my computer, but did not solve the problem

**Comment by GantMan** [comment]
I have a terrible fix.  

I guessed that the problem was some kind of dependency upgrade, and because React Native does not come locked to working versions, something got published and broke it.

I grabbed a yarn.lock from a project from over the weekend, pasted that into my project's `yarn.lock` file and then did `yarn` at the command line to have it automatically prune/lock things back up.   Somewhere in that mess, it grabbed the right stuff and now I can use Android again.   Not a great fix, but hopefully it helps!

### TL:DR;
1. Copy this `yarn.lock` contents https://gist.github.com/GantMan/347fb4206fa7be141ff7133ba4859afd
2. Past it over your project's `yarn.lock` contents
3. run `yarn` in the CLI

**Comment by sunnylqm** [comment]
@GantMan I just did the same thing. I saw mainly babel update when diffing a working version and a broken version. It is hard to tell what deps cause the break, so the best option is definitely to use a whole working lock file.

**Comment by rolignu2** [comment]
please , reopen the issue. i have the same problem it probably because the object.getOwnPropertiesDescriptor doesn't exists anymore, the same happended on 
react-native-keyboard-aware-scroll-view, the problem is on 

node_modules/@babel/runtime/helpers/objectSpread.js 

Object.defineProperties(target, Object.getOwnPropertyDescriptors(arguments[i]));

but i don't  know how i can solve

**Comment by ashishmundra** [comment]
same issue here

**Comment by sunnylqm** [comment]
@rolignu2 Thanks for locating the line, so I guess [this pr of babel-helpers](https://github.com/babel/babel/commit/3ae5e79ec8ab1e6d1c6927f35d4a1eeac04faf9a#diff-ba75bd4d116cc90ad8c8002e3010e821) cause the problem. It forgot to check this function in else branch. Please fix this asap  @hzoo @saschanaz

**Comment by mgwidmann** [comment]
@sunnylqm Wow merged 10 hours ago... Sad that the NPM world is like this.

https://github.com/babel/babel/pull/9341

**Comment by rolignu2** [comment]
@sunnylqm thanks man to apply the revert.

**Comment by Army-92** [comment]
@rolignu2 Thanks for locating the line
I refer to the project I used before.
It is work for me.
node_modules/@babel/runtime/helpers/objectSpread.js
Replace with the code below in the path above.

var defineProperty = require("./defineProperty");

function _objectSpread(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {};
    var ownKeys = Object.keys(source);

    if (typeof Object.getOwnPropertySymbols === 'function') {
      ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) {
        return Object.getOwnPropertyDescriptor(source, sym).enumerable;
      }));
    }

    ownKeys.forEach(function (key) {
      defineProperty(target, key, source[key]);
    });
  }

  return target;
}

module.exports = _objectSpread;

**Comment by sunnylqm** [comment]
Babel just published a hotfixed version. Just remove your yarn.lock and `yarn` and this issue should be gone.

**Comment by bepehr** [comment]
@sunnylqm solution worked for me , just add that lines to package.json , and run yarn then run react-native run-android

**Comment by raghulvaikundam** [comment]
run npm install in the project directory to fix the issue. This command will install the missed dependency for  the react native project

**Comment by ChhXin** [comment]
> run npm install in the project directory to fix the issue. This command will install the missed dependency for the react native project

i tried it, it didn't work...

**Comment by ChhXin** [comment]
Retry `react-native init name`，it's ok!

**Comment by raghulvaikundam** [comment]
> > run npm install in the project directory to fix the issue. This command will install the missed dependency for the react native project
> 
> i tried it, it didn't work...

maybe u check with your package.json. Tried with latest react-native version 0.57

**Comment by dffalla** [comment]
I solve this proplem now

1-Delete the old project and create a new project
 2-Then open the cmd for the project path and type
3-npm istall
Then
4-react-native init AwesomeProject

**Comment by kelset** [comment]
Hey everyone, looks like this was a Babel issue - they should have published an hotfix on their side already, in the meantime to avoid too much noise I'll lock this issue.

A workaround can be [found here](https://github.com/facebook/react-native/issues/23092) - thanks @sunnylqm 👍
